### ISSUE TITLE
Jackson 2.13.0 compatibility

### ISSUE THREAD

**Issue by ben-manes** [json-path/JsonPath#778]
When upgrading this dependency and using the jackson backend, I noticed that JsonPath no longer worked. The path expressions, including a simple `$`, all returned null values. I simply reverted back to `2.12.5` which works fine. I have not debugged far enough to determine if it is a JsonPath or Jackson compatibility bug.

/cc @cowtowncoder

**Comment by cowtowncoder** [comment]
I cannot think of specific change (and obviously breakages in general are not by design), but if someone can reproduce the issue easily, I'd be happy to help figure it out -- and possibly fix issue, if possible, for upcoming 2.13.1 patch.

**Comment by ben-manes** [comment]
I wrote a [minimal project](https://github.com/ben-manes/jsonpath-jackson) which demonstrates this failure. Simply change the Jackson version in the [build.gradle](https://github.com/ben-manes/jsonpath-jackson/blob/317c10b3b15e1463ad23219c5076fb0c0e741b7a/build.gradle#L11-L12) and run `./gradlew test`. It will fail with the newest version.

This fails only when `DeserializationFeature.FAIL_ON_TRAILING_TOKENS` is enabled, so I think this might be your bug @cowtowncoder, wdyt?

**Comment by ben-manes** [comment]
Oh right, I suppressed exceptions in order to have it return null when an absent value. When that is changed in `JsonPathDefaults`,

```json
{
  "class": "JsonPathTest",
  "method": "read"
}
```

```java
FAILED: read
java.lang.IllegalArgumentException: Trailing token (of type VALUE_STRING) found after value (bound as `com.fasterxml.jackson.databind.JsonNode`): not allowed as per `DeserializationFeature.FAIL_ON_TRAILING_TOKENS`
 at [Source: UNKNOWN; byte offset: #UNKNOWN]
	at com.fasterxml.jackson.databind.ObjectMapper.valueToTree(ObjectMapper.java:3395)
	at com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider.createJsonElement(JacksonJsonNodeJsonProvider.java:235)
	at com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider.setArrayIndex(JacksonJsonNodeJsonProvider.java:138)
	at com.jayway.jsonpath.internal.path.EvaluationContextImpl.addResult(EvaluationContextImpl.java:81)
	at com.jayway.jsonpath.internal.path.PathToken.handleObjectProperty(PathToken.java:78)
	at com.jayway.jsonpath.internal.path.PropertyPathToken.evaluate(PropertyPathToken.java:79)
	at com.jayway.jsonpath.internal.path.RootPathToken.evaluate(RootPathToken.java:62)
	at com.jayway.jsonpath.internal.path.CompiledPath.evaluate(CompiledPath.java:99)
	at com.jayway.jsonpath.internal.path.CompiledPath.evaluate(CompiledPath.java:107)
	at com.jayway.jsonpath.JsonPath.read(JsonPath.java:185)
	at com.jayway.jsonpath.internal.JsonContext.read(JsonContext.java:89)
	at com.jayway.jsonpath.internal.JsonContext.read(JsonContext.java:78)
	at com.jayway.jsonpath.internal.JsonContext.read(JsonContext.java:83)
	at jsonpath.jackson.JsonPathTest.read(JsonPathTest.java:33)
```

**Comment by cowtowncoder** [comment]
@ben-manes Possible, given that `FAIL_ON_TRAILING_TOKENS` is relatively new feature. But I'd have to get an isolated test to show kind of input is given -- looks like this is with buffering, and with buffers content may behave differently than at main level. It is possible that this is an issue with usage of `ObjectMapper` methods as well (i.e. something JsonPath may need to do differently) -- this could be the case if a check for trailing tokens was formerly missing.

So difficult to say yet.

**Comment by kallestenflo** [comment]
I just released json-path-1.7.0 and bumped Jackson version to 2.13.1. Your test case now runs as expected. Maybe a problem with Jackson 2.13.0?

**Comment by ben-manes** [comment]
Thanks @kallestenflo and @cowtowncoder! It looks like I ran into https://github.com/FasterXML/jackson-databind/issues/3308. I appreciate the support, as I know this OSS stuff is all done in the limited free time (and energy!) outside of work, family, etc.
